client/src/hooks/Provider.tsx

```diff
@@ -92,8 +105,29 @@ export default class StateProvider extends React.Component<Props, State> {
     return value;
   }
 
+  async apiPut<P extends MinimumApiObject>(endpoint: ApiEndpoint, value: P): Promise<void> {
```

Comment:
Part of me wants this to not be called "endpoint", and instead something like "resource" -- I think of endpoint as a full path rather than a parameter to a path.

Reply:
See the latest, changed all endpoint to resource